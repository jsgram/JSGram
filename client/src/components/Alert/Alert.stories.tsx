import React from 'react';
import { storiesOf } from '@storybook/react';
import { clearAlert } from '../../store/alert/actions';

import Alert from './index';

export const alertState = {
    message: 'Comment added successfully',
    color: 'success',
    clearAlert,
};

storiesOf('Alert', module)
    .add('success',
        () => <Alert {...alertState}/>,
        )
    .add('error',
        () => <Alert {...alertState} message='Error' color='danger'/>,
        );
