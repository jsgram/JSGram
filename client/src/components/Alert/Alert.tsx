import React, { ReactElement } from 'react';
import { UncontrolledAlert } from 'reactstrap';

interface IAlertProps {
    clearAlert: any;
    message: string;
    color: string;
}

const Alert: React.FunctionComponent<IAlertProps> = ({
  clearAlert,
  message,
  color,
}: {clearAlert: () => void, message: string, color: string}): ReactElement | null => message ?
    <UncontrolledAlert color={color} onClick={(): void => { clearAlert(); }}>{message}</UncontrolledAlert> : null;

export default Alert;
