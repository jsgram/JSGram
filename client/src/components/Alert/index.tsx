import React from 'react';
import { FormProps, UncontrolledAlert } from 'reactstrap';
import { clearAlert } from '../../store/alert/actions';
import './style.scss';

export default class Alert extends React.Component<FormProps> {
    public timeout: any = null;

    public componentDidMount(): void {
        this.timeout = setTimeout(() => {
            this.props.clearAlert();
        }, 3000);
    }

    public componentWillUnmount(): void {
        clearTimeout(this.timeout);
        this.timeout = 0;
    }

    public render(): JSX.Element {
        const {message, color}: FormProps = this.props;
        return message && (
            <UncontrolledAlert
                color={color}
                className='position-fixed'
                onClick={(): void => {
                    clearAlert();
                }}>{message}
            </UncontrolledAlert>
        );
    }
}
