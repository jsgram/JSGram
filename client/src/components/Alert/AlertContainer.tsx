import React from "react";
import { connect } from "react-redux";
import { clearAlert } from "../../store/alert/actions";
import Alert from "./Alert";

interface AlertProps {
  clearAlert: Function;
  message: string;
  color: string;
}

interface Alert {
  message: string;
  color: string;
}

interface AlertState {
  alert: Alert;
}

const AlertContainer: React.FunctionComponent<AlertProps> = ({
  message,
  color,
  clearAlert
}) => <Alert clearAlert={clearAlert} message={message} color={color} />;

const mapStateToProps = (state: AlertState) => ({
  message: state.alert.message,
  color: state.alert.color
});

export default connect(
  mapStateToProps,
  { clearAlert }
)(AlertContainer);
