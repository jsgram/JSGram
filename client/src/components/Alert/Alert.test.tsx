import Alert from './Alert';
import * as reactstrap from 'reactstrap';

import { shallow } from 'enzyme';
import React from 'react';

describe('Alert component:', () => {
    const props = {
        message: 'somemessage',
        color: 'somecolor',
        clearAlert: (): string => 'somehandler',
    };
    let renderer;

    beforeEach(() => {
        reactstrap.UncontrolledAlert = jest.fn(() => (<div>{props.message}</div>));
        renderer = shallow(<Alert {...props} />);
    });

    test('componentDidMount - success', () => {
        expect(renderer.instance().timeout).toBe(2);
    });

    test('componentWillUnmount - success', () => {
        renderer.instance().componentWillUnmount();
        expect(renderer.instance().timeout).toBe(0);
    });

    test('render - success', () => {
        expect(renderer.html()).toBe('<div>somemessage</div>');
    });
});
