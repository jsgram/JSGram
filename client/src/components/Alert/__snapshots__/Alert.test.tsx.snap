// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component: render - success 1`] = `ShallowWrapper {}`;
